Add Go database query builder library for PostgreSQL

arthurkushman opened the pull request:
[![Go Report Card](https://goreportcard.com/badge/github.com/arthurkushman/buildsqlx)](https://goreportcard.com/report/github.com/arthurkushman/buildsqlx)
[![Build and run](https://github.com/arthurkushman/buildsqlx/workflows/Build%20and%20run/badge.svg)](https://github.com/arthurkushman/buildsqlx/actions)
[![GoDoc](https://github.com/golang/gddo/blob/c782c79e0a3c3282dacdaaebeff9e6fd99cb2919/gddo-server/assets/status.svg)](https://godoc.org/github.com/arthurkushman/buildsqlx)
[![codecov](https://codecov.io/gh/arthurkushman/buildsqlx/branch/master/graph/badge.svg)](https://codecov.io/gh/arthurkushman/buildsqlx)
[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)

- github.com repo: https://github.com/arthurkushman/buildsqlx
- godoc.org: https://godoc.org/github.com/arthurkushman/buildsqlx
- goreportcard.com: https://goreportcard.com/report/github.com/arthurkushman/buildsqlx
- coverage service link: https://codecov.io/gh/arthurkushman/buildsqlx

avelino commented:
Thank you for contributing with [awesome-go](https://github.com/avelino/awesome-go), we will revise your contribution as soon as possible.

Automation body content check:
- godoc.org: True
- goreportcard.com: True
- coverage: True

arthurkushman commented:
Hi, fixed all possible linter issues, but those 3 cannot be fixed due to the linter bug:
```
buildsqlx/schema.go
Line 252: warning: comment on exported method Table.Text should be of the form "Text ..." (golint)
Line 258: warning: comment on exported method Table.DblPrecision should be of the form "DblPrecision ..." (golint)
Line 326: warning: comment on exported method Table.Date should be of the form "Date ..." (golint)
```
It doesn't recognize Text/Date names of functions  and somehow DblPrecision

Also fixed ineffassign, but it is buggy with the most popular tool for testing in Go here:
```
Line 249: warning: ineffectual assignment to rows (ineffassign)
Line 321: warning: ineffectual assignment to err (ineffassign)
Line 624: warning: ineffectual assignment to err (ineffassign)
Line 627: warning: ineffectual assignment to err (ineffassign)
Line 638: warning: ineffectual assignment to err (ineffassign)
```

For example these lines of code is totally normal:
```go
rows, err := db.Table(TestTable).Replace(obj.insert, obj.conflict)
assert.NoError(t, err)
```
Nevertheless, those checks are green now, because of some lucky fixes

arthurkushman commented:
> The build status is failing, please fix it.

It is quite difficult to fix, because it is rather a GitHub Actions problem - configs is just the same as succeeded ones, but are not working properly with network bound from PostgreSQL service. Anyway, I'll try to fix if find the solution.
